src/library/scala/collection/TraversableLike.scala

```diff
@@ -606,12 +606,21 @@ trait TraversableLike[+A, +Repr] extends Any
    *           simple name of the collection class $coll.
    */
   def stringPrefix : String = {
-    var string = repr.getClass.getName
-    val idx1 = string.lastIndexOf('.' : Int)
-    if (idx1 != -1) string = string.substring(idx1 + 1)
-    val idx2 = string.indexOf('$')
-    if (idx2 != -1) string = string.substring(0, idx2)
-    string
+    val fqn = repr.getClass.getName
+    val cls = {
+      val idx1 = fqn.lastIndexOf('.' : Int)
```

Comment:
nitpick (carried over from the old code): the `: Int` is doing nothing here

doesn't this part boil down to `val cls = fqn.split('.').last`?